doozer/doozerlib/cli/images_health.py

```diff
@@ -1,189 +1,136 @@
-
-    # Use a regular expression to remove single quotes from the start and end of lines starting with a hyphen
-    pattern = re.compile(r"^-\s+'(.*?)'$", re.MULTILINE)
-    modified_yaml_output = pattern.sub(r"- \1", yaml_output)
-
-    print(modified_yaml_output)
+        # Filter out results without concerns
+        concerns = {k: v for k, v in self.concerns.items() if v}
 
+        # We should now have a dict of qualified_key => [concern, ...]
+        if not concerns:
+            logging.getLogger(__name__).info('No concerns to report!')
+        click.echo(json.dumps(concerns, indent=4))
 
-def get_concerns(image, runtime, limit, url_markup):
-    image_concerns = []
-
-    def add_concern(msg):
-        image_concerns.append(msg)
-
-    def url_text(url, text):
-        if url_markup == 'slack':
+    def url_text(self, url, text):
+        if self.url_markup == 'slack':
             return f'<{url}|{text}>'
-        if url_markup == 'github':
+        if self.url_markup == 'github':
             return f'[{text}]({url})'
-        raise IOError(f'Unknown markup mode: {url_markup}')
-
-    records = query(image, runtime, limit)
-
-    if not records:
-        add_concern('Image build has never been attempted')
-        return image_concerns
-
-    latest_success_idx = -1
-    latest_success_bi = None
-    latest_success_bi_task_url = ''
-    latest_success_bi_build_url = ''
-    latest_success_bi_dt = ''
-
-    for idx, record in enumerate(records):
-        if record[1] == 'success':
-            latest_success_idx = idx
-            latest_success_bi = record
-            latest_success_bi_task_url = f"{BREWWEB_URL}/taskinfo?taskID={latest_success_bi[0]}"
-            latest_success_bi_build_url = latest_success_bi[3]
-            latest_success_bi_dt = time.strftime("%Y-%m-%d %H:%M:%S", time.localtime(latest_success_bi[2] / 1000))
-            break
-
-    latest_attempt_build_url = records[0][3]
-    latest_attempt_task_url = f"{BREWWEB_URL}/taskinfo?taskID={records[0][0]}"
-    oldest_attempt_bi_dt = time.strftime("%Y-%m-%d %H:%M:%S", time.localtime(records[-1][2] / 1000))
-
-    # Generate the Art-Dash link
-    art_dash_link = generate_art_dash_history_link(image, runtime)
-
-    if latest_success_idx != 0:
-        msg = f'Latest attempt {url_text(latest_attempt_task_url, "failed")} ({url_text(latest_attempt_build_url, "jenkins job")}); '
-
-        # The latest attempt was a failure
-        if latest_success_idx == -1:
-            # No success record was found
-            msg += f'Failing for at least the last {len(records)} attempts / {oldest_attempt_bi_dt}'
+        raise IOError(f'Unknown markup mode: {self.url_markup}')
+
+    async def get_concerns(self, image_meta):
+        key = image_meta.distgit_key
+        self.concerns[key] = []
+
+        builds = await self.query(image_meta)
+
+        if not builds:
+            self.concerns[key].append(f'Image build has never been attempted during last {DELTA_DAYS} days')
```

Comment:
Think we do not need to add it to concerns. No build in 7 days is normal for a slow release/rhel version.

Reply:
Agreed and updated